src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionSettings.cs

```diff
@@ -31,13 +28,29 @@ internal sealed class HttpConnectionSettings
         internal int _maxConnectionsPerServer = HttpHandlerDefaults.DefaultMaxConnectionsPerServer;
         internal int _maxResponseHeadersLength = HttpHandlerDefaults.DefaultMaxResponseHeadersLength;
 
-        internal ClientCertificateOption _clientCertificateOptions = HttpHandlerDefaults.DefaultClientCertificateOption;
-        internal X509CertificateCollection _clientCertificates;
-
-        internal Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> _serverCertificateCustomValidationCallback;
-        internal bool _checkCertificateRevocationList = false;
-        internal SslProtocols _sslProtocols = SslProtocols.None;
+        internal SslClientAuthenticationOptions _sslOptions;
 
         internal IDictionary<string, object> _properties;
+
+        public HttpConnectionSettings Clone() =>
+            new HttpConnectionSettings()
+            {
+                _automaticDecompression = _automaticDecompression,
+                _useCookies = _useCookies,
+                _cookieContainer = _cookieContainer,
+                _useProxy = _useProxy,
+                _proxy = _proxy,
+                _defaultProxyCredentials = _defaultProxyCredentials,
+                _preAuthenticate = _preAuthenticate,
+                _credentials = _credentials,
+                _allowAutoRedirect = _allowAutoRedirect,
+                _maxAutomaticRedirections = _maxAutomaticRedirections,
+                _maxConnectionsPerServer = _maxConnectionsPerServer,
+                _maxResponseHeadersLength = _maxResponseHeadersLength,
+                _sslOptions = _sslOptions?.ShallowClone(), // shallow clone the security options for basic prevent of mutation issues while processing
+                _properties = _properties
+            };
+            // TODO: Should _properties, _cookieContainer, ClientCertificates be deep clones rather than shallow?
+            // They're not today, which means changes are visible in both directions (and not thread-safe).
```

Comment:
I plan to just delete this TODO, but let me know if you think we should do something different here.  In general the "don't change after use" detection seems to just be intended as a minimum help and not something we need to really bend over backwards for.

Reply:
The concern that the settings would be changed on the option bag for TLS was discussed ad nauseam in the API design and it was decided I think that no other API cares if you do something silly. That and if you have passed it into SslStream, it copies inside anyway, so it won't change its behaviour...